src/FileMigrationHelper.php

```diff
@@ -56,27 +61,40 @@ public function run($base = null)
         Environment::increaseTimeLimitTo();
         Environment::increaseMemoryLimitTo();
 
+        $this->logger = Injector::inst()->get(LoggerInterface::class);
+
         // Loop over all files
-        $count = 0;
+        $i = $migrated = 0;
         $originalState = null;
         if (class_exists(Versioned::class)) {
             $originalState = Versioned::get_reading_mode();
             Versioned::set_stage(Versioned::DRAFT);
         }
 
-        foreach ($this->getFileQuery() as $file) {
+        $query = $this->getFileQuery();
+        $total = $query->count();
+        foreach ($query as $file) {
             // Bypass the accessor and the filename from the column
             $filename = $file->getField('Filename');
 
             $success = $this->migrateFile($base, $file, $filename);
+
+            if ($i % 100 == 0) {
+                /** @var LoggerInterface $logger */
+                if ($this->logger != null) {
+                    $this->logger->info("Iterated $i out of $total files. Migrated $migrated files.");
```

Comment:
Personally I think this line could just read `Injector::inst()->get(LoggerInterface::class)->info([...]);`. Injector throws an exception if the service cannot be found (and the earlier call wasn't handling that either). This saves adding a private variable and the checks.

Reply:
nice catch!
though, I think we don't want to abort the whole thing in case the logger hasn't been registered

Reply:
Chiming in to say that currently there'll always be a LoggerInterface defined (by core). In future (hopefully) we'll have it separated so that LoggerInterface by default is a user-defined logger, which will also exist (see https://github.com/silverstripe/silverstripe-framework/pull/8241). When that PR gets merged it should have a NullHandler by default on the LoggerInterface.

I'm fine with having the logger injected via Injector DI, you shouldn't need the truthy check though, but I'm mostly just mentioning this for reference - don't consider this a review =)